app/main.py

```diff
@@ -1,2 +1,87 @@
+from typing import Mapping, Any, Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = 8, load_factor: float = 0.75) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.count = 0
+        self.table = [None] * self.capacity
+
+    def __iter__(self) -> Any:
+        for item in self.table:
+            if item is not None:
+                yield item[0]
+
+    def __len__(self) -> int:
+        return self.count
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.count / self.capacity >= self.load_factor:
+            self._resize()
+
+        index = self._hash(key) % self.capacity
+        while self.table[index] is not None and self.table[index][0] != key:
+            index = (index + 1) % self.capacity
+
+        if self.table[index] is None:
+            self.count += 1
+            self.table[index] = (key, self._hash(key), value)
+        else:
+            self.table[index] = (key, self.table[index][1], value)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._find_key(key)
+        if index is None:
+            raise KeyError(key)
+        else:
+            return self.table[index][2]
+
+    def pop(self, key: object = None, default: Any = None) -> Any:
+        if key is None:
+            raise ValueError(
+                "pop() method with `Dictionary` works only with defined key"
+            )
+
+        index = self._find_key(key)
+        if index is None:
+            return default
+        else:
+            value = self.table[index][2]
+            self.table[index] = None
+            self.count -= 1
+            return value
+
+    def update(self, other: Mapping[Hashable, Any]) -> None:
+        for key, value in other.items():
+            self[key] = value
```

Comment:
![image](https://user-images.githubusercontent.com/105493988/221921937-5424c245-0541-4e4f-a489-bcba8b28652d.png)

Reply:
I'm sorry, but where you found that info? Here is my: https://docs.python.org/3.11/library/stdtypes.html#dict.update
![image](https://user-images.githubusercontent.com/107141441/221926228-2ef2ed84-a45c-4fbc-908c-35daaec2c3c1.png)

Reply:
Your source is correct, but update() for dict can also take `tuples`, for example. And tuples don't have `.items()`